index.js

```diff
@@ -0,0 +1,487 @@
+let cartProducts = [];
+let brandSet = new Set();

```

Comment:
The approach with so many state will cause lot of problems. Try to have only cart and products state.